app/controller/auth/signup.py

```diff
@@ -0,0 +1,57 @@
+from fastapi import APIRouter, Body
+
+from app.dto.model.customer import SignupDTO, EmailVerificationDTO
+
+router = APIRouter(prefix="/api/v1/auth")
+
+
+@router.post('/signup')
+async def signup(data: SignupDTO = Body(...)):
+    from app.service.model.customer import CustomerLib
+    from fastapi.encoders import jsonable_encoder
+    from app.config.success_response import SuccessResponse, CustomException
+
+    # check email exist
+    customer = CustomerLib.find_by(where={"email": data.email})
+    if customer:
+        from fastapi import status
+
+        return CustomException(error="email Already Exist", status_code=status.HTTP_400_BAD_REQUEST)
+
+    # create user account (includes password hashing)
+    new_customer = CustomerLib.signup_user(data=jsonable_encoder(data))
+
+    print(new_customer)
+
+    # send email
+    await CustomerLib.send_emails(new_customer)
+
+    return SuccessResponse(data=new_customer)
+
+
+@router.post("/verify-email")
```

Comment:
create a new py file in auth and call it emial_verification.py and move this to it -> it's one file per route